### PR TITLE
Restart after update

### PR DESCRIPTION
**Pull request by wingio**

### CODE REVIEW

**Comment by Vendicated** [comment]
didnt u say this dioesnt work on some phones

**Comment by wingio** [comment]
> didnt u say this dioesnt work on some phones

No, the problem with zt's plugin was just the button not showing up

File: Aliucord/src/main/java/com/aliucord/settings/Updater.java [hunk 1]
```diff
@@ -155,12 +164,18 @@ public void onViewBound(View view) {
...
 
             //noinspection ConstantConditions
             refreshButton.setImageDrawable(Utils.tintToTheme(ContextCompat.getDrawable(context, R.d.ic_refresh_white_a60_24dp).mutate()), false);
             updateAllButton.setImageDrawable(ContextCompat.getDrawable(context, R.d.ic_file_download_white_24dp));
             settingsButton.setImageDrawable(ContextCompat.getDrawable(context, R.d.ic_guild_settings_24dp));
 
+            Drawable restartDrawable = ContextCompat.getDrawable(context, R.d.ic_refresh_white_a60_24dp).mutate();
+            restartDrawable.setTint(0xff57F287);
+            restartButton.setImageDrawable(restartDrawable);
+            restartButton.setVisibility(showRestartButton ? View.VISIBLE : View.GONE);
```
Comment (Vendicated) [inline Aliucord/src/main/java/com/aliucord/settings/Updater.java:177]:
```suggestion
            restartButton.setVisibility(View.GONE);
```

File: Aliucord/src/main/java/com/aliucord/settings/Updater.java [hunk 2]
```diff
@@ -122,6 +127,8 @@ public void onViewBound(View view) {
                                 try {
                                     updateAliucord(ctx);
                                     Utils.showToast(ctx, "Successfully updated Aliucord. Please restart Aliucord to load the update!");
+                                    showRestartButton = true;
+                                    Utils.mainThread.post(this::reRender);
```
Comment (Vendicated) [inline Aliucord/src/main/java/com/aliucord/settings/Updater.java:131]:
```suggestion
                                    getHeaderBar().findViewById(restartId).setVisibility(View.VISIBLE);
```

(and also add code to generate & add that id to the correct button)

File: Aliucord/src/main/java/com/aliucord/updater/Updater.java [hunk 3]
```diff
@@ -6,12 +6,14 @@
...
 
 import android.content.Context;
+import android.content.Intent;
 
 import com.aliucord.*;
 import com.aliucord.utils.ReflectUtils;
 
 import java.io.File;
 import java.io.IOException;
+import java.lang.Runtime;
```
Comment (Vendicated) [inline Aliucord/src/main/java/com/aliucord/updater/Updater.java:16]: unused imports?

**Review by Vendicated — DISMISSED** [review]

File: Aliucord/src/main/java/com/aliucord/settings/Updater.java (hunk 2 above)
Comment (wingio) [reply Aliucord/src/main/java/com/aliucord/settings/Updater.java:131]: only reason i didn't do this was bc i thought you couldn't edit views on a different thread from the one they were created in

Comment (Vendicated) [reply Aliucord/src/main/java/com/aliucord/settings/Updater.java:131]: post it to the main thread then :P but it should work i think idk just try lol

File: Aliucord/src/main/java/com/aliucord/updater/Updater.java [hunk 4]
```diff
@@ -6,6 +6,7 @@
 package com.aliucord.updater;
 
 import android.content.Context;
+import android.content.Intent;
```
Comment (Juby210) [inline Aliucord/src/main/java/com/aliucord/updater/Updater.java:9]: unused import
```suggestion
```

File: Aliucord/src/main/java/com/aliucord/settings/Updater.java [hunk 5]
```diff
@@ -13,7 +13,9 @@
 
 import android.annotation.SuppressLint;
 import android.content.Context;
+import android.content.Intent;
 import android.graphics.Color;
+import android.graphics.drawable.Drawable;
```
Comment (Juby210) [inline Aliucord/src/main/java/com/aliucord/settings/Updater.java:18]: unused import
```suggestion
```

File: Aliucord/src/main/java/com/aliucord/settings/Updater.java [hunk 6]
```diff
@@ -121,7 +125,18 @@ public void onViewBound(View view) {
...
                                     updateAliucord(ctx);
-                                    Utils.showToast(ctx, "Successfully updated Aliucord. Please restart Aliucord to load the update!");
+                                    Utils.showToast(ctx, "Successfully updated Aliucord.");
+                                    Snackbar rb = Snackbar
+                                    .make(getLinearLayout(), "Restart to apply the update.", Snackbar.LENGTH_INDEFINITE)
+                                    .setAction("Restart", e -> {
+                                        Intent intent = context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
+                                        context.startActivity(Intent.makeRestartActivityTask(intent.getComponent()));
+                                        Runtime.getRuntime().exit(0);
+                                    });
```
Comment (Juby210) [inline Aliucord/src/main/java/com/aliucord/settings/Updater.java:135]: adjust code style
```suggestion
                                        .make(getLinearLayout(), "Restart to apply the update.", Snackbar.LENGTH_INDEFINITE)
                                        .setAction("Restart", e -> {
                                            Intent intent = context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
                                            context.startActivity(Intent.makeRestartActivityTask(intent.getComponent()));
                                            Runtime.getRuntime().exit(0);
                                        });
```

**Review by Juby210 — CHANGES_REQUESTED** [review]

**Comment by wingio** [comment]
Ok i applied your suggestions @Juby210

**Review by Juby210 — APPROVED** [review]

**Pull request merged**